### PR TITLE
[kube-prometheus-stack] bump grafana 6.37.1 (9.1.2)

### PR DESCRIPTION
**Pull request by andrejshapal**
Signed-off-by: andrejshapal <96792836+andrejshapal@users.noreply.github.com>

Updated grafana to helm version 6.37.1  and app version 9.1.2

### CODE REVIEW

**Comment by andrejshapal** [comment]
Hello, 
Really sorry if I did something wrong. Just looked in previous similar prs how grafana is getting bumped and tried to do the same.

**Review by monotek — CHANGES_REQUESTED** [review]
Please raise chart version too

**Comment by Allex1** [comment]
@andrejshapal you need to update the dependencies in Chart.yaml and then do a `helm dep update` in order for the Chart.lock to get updated. Can you please update the Grafana to latest ? 6.37.*

**Comment by andrejshapal** [comment]
> @andrejshapal you need to update the dependencies in Chart.yaml and then do a `helm dep update` in order for the Chart.lock to get updated. Can you please update the Grafana to latest ? 6.37.*

Thank you. Have updated version in chart.yaml and run `helm dep update`.

**Comment by andrejshapal** [comment]
If I understand correctly from the error was in lint test "Chart is not ok. Need to bump version", it was due to chart version was not changed. The last commit is correct?

**Comment by zanhsieh** [comment]
@andrejshapal 
I would suggest you bump to Grafana chart version: 6.37.1 appVersion: 9.1.2 due to CVE:

https://grafana.com/blog/2022/08/30/security-release-new-versions-of-grafana-and-grafana-image-renderer-with-a-high-severity-security-fix-for-cve-2022-31176/

**Comment by andrejshapal** [comment]
@zanhsieh Bumped to this one, will update title.

**Review by andrejshapal — COMMENTED** [review]
@monotek Should not you review this?

**Comment by Allex1** [comment]
Maybe we should split this pr in 2: 
- 1 for the kube-prometheus-stack version bumps which should also include kube-state-metrics dependencies bumps
- 1 for kube-state-metrics bumps.

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 39.11.0
+version: 39.11.1
```
Comment (dotdc) [inline charts/kube-prometheus-stack/Chart.yaml:25]: This should be 39.12.0 because you bump minors in both sub charts.

**Review by dotdc — CHANGES_REQUESTED** [review]

**Comment by andrejshapal** [comment]
> Maybe we should split this pr per chart:
> 
> * 1 for the kube-prometheus-stack version bumps which should also include kube-state-metrics dependencies bumps
> * 1 for kube-state-metrics bumps.

You mean one pr for grafana and one for  kube-state-metrics? Or I did not get it.

**Comment by dotdc** [comment]
> > Maybe we should split this pr per chart:
> > 
> > * 1 for the kube-prometheus-stack version bumps which should also include kube-state-metrics dependencies bumps
> > * 1 for kube-state-metrics bumps.
> 
> You mean one pr for grafana and one for kube-state-metrics? Or I did not get it.

Just checked, only the change for `kube-prometheus-stack` is needed. 
You just need to drop your changes in `kube-state-metrics` in this PR because the chart is already updated: https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-state-metrics/Chart.yaml

**Comment by dotdc** [comment]
@andrejshapal You delete the entire files in https://github.com/prometheus-community/helm-charts/pull/2431/commits/39954a859d08b67c1b387dae99f9e4fab6dc8106 and https://github.com/prometheus-community/helm-charts/pull/2431/commits/0f156a47adb92dd0725f61ef4803ce296ef70058, this is not good, just revert your changes.

**Comment by andrejshapal** [comment]
@dotdc Yeah, I already searching revert button)

**Comment by dotdc** [comment]
The code seems good now, you just need to have the DCO right: 
- https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md
- https://github.com/prometheus-community/helm-charts/pull/2431/checks?check_run_id=8266798972

**Comment by andrejshapal** [comment]
> The code seems good now, you just need to have the DCO right:
> 
> * https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md
> * https://github.com/prometheus-community/helm-charts/pull/2431/checks?check_run_id=8266798972

Yep, a bit messy, but fixed it in the end of the end.

**Comment by dotdc** [comment]
Not a big deal because it will be squash in the end, but it's generally better if you do it in your branch :wink:

**Review by dotdc — APPROVED** [review]

**Comment by dotdc** [comment]
LGTM :+1: 
You just need to wait for a `kube-prometheus-stack` maintainer to accept your PR.

**Comment by andrejshapal** [comment]
@dotdc Many thanks. Have a good rest of the day ;)

**Review by zanhsieh — APPROVED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**